test/e2e/common/pods.go

```diff
@@ -848,4 +853,115 @@ var _ = framework.KubeDescribe("Pods", func() {
 		validatePodReadiness(false)
 
 	})
+
+	ginkgo.It("should run through the lifecycle of Pods and PodStatus", func() {
+		podResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+		testNs := "default"
+		testPodName := "pod-test"
+		testPodImage := "nginx"
+		testPodImage2 := "nginx"
+		testPod := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testPodName,
+				Labels: map[string]string{
+					"test-pod-static": "true",
+				},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  testPodName,
+						Image: testPodImage,
+					},
+				},
+			},
+		}
+		ginkgo.By("creating a Pod with a static label")
+		_, err := f.ClientSet.CoreV1().Pods(testNs).Create(context.TODO(), &testPod, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create Pod")
+
+		ginkgo.By("setting up a watch for the Pod")
+		podWatchTimeoutSeconds := int64(180)
+		podWatch, err := f.ClientSet.CoreV1().Pods(testNs).Watch(context.TODO(), metav1.ListOptions{LabelSelector: "test-pod-static=true", TimeoutSeconds: &podWatchTimeoutSeconds})
+		framework.ExpectNoError(err, "failed to set up watch")
+
+		podWatchChan := podWatch.ResultChan()
+		ginkgo.By("watching for Pod to be ready")
+		for watchEvent := range podWatchChan {
+			podWatchEvent, ok := watchEvent.Object.(*v1.Pod)
+			framework.ExpectEqual(ok, true, "unable to fix type")
+			if podWatchEvent.Status.Phase == "Running" {
+				break
+			}
+		}
+
+		ginkgo.By("patching the Pod with a new Label and updated data")
+		podPatch, err := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]string{
+					"podtemplate": "patched",
+				},
+			},
+			"spec": map[string]interface{}{
+				"containers": []map[string]interface{}{
+					{
+						"name":  testPodName,
+						"image": testPodImage2,
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err, "failed to marshal JSON patch for Pod")
+		_, err = f.ClientSet.CoreV1().Pods(testNs).Patch(context.TODO(), testPodName, types.StrategicMergePatchType, []byte(podPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch Pod")
+
+		ginkgo.By("getting the Pod and ensure it's patched")
```

Comment:
match tense of verbs:
`getting the Pod and ensuring it's patched`

Reply:
Updated in 1e98c1ba6c0